### ISSUE TITLE
Broken regex redirects

### ISSUE THREAD

**Issue by raptureaid** [nystudio107/craft-retour#107]
Following the 3.1.34/3.1.35 releases, our regex redirects stopped resolving correctly.

![image](https://user-images.githubusercontent.com/5720326/77089344-1a1c3800-69dc-11ea-86c5-88e11dc6011f.png)

Results in: `https://www.lynn.edu/http:/lynn-library.libguides.com/library/home`

Pinning composer to the 3.1.33 release results in the redirects working as expected.

**Comment by dlindberg** [comment]
I’m seeing the same issue, the url helper seems to be throwing an issue:

```
Copy Stacktrace Search Stackoverflow Search Google Error
PHP Fatal Error – yii\base\ErrorException

Uncaught TypeError: Argument 4 passed to craft\helpers\UrlHelper::siteUrl() must be of the type integer or null, string given, called in /home/forge/deathpenaltyinfo.org/vendor/nystudio107/craft-retour/src/services/Redirects.php on line 221 and defined in /home/forge/deathpenaltyinfo.org/vendor/craftcms/cms/src/helpers/UrlHelper.php:288
Stack trace:
#0 /home/forge/deathpenaltyinfo.org/vendor/nystudio107/craft-retour/src/services/Redirects.php(221): craft\helpers\UrlHelper::siteUrl('https:/files.de...', NULL, NULL, '')
#1 /home/forge/deathpenaltyinfo.org/vendor/nystudio107/craft-retour/src/services/Redirects.php(178): nystudio107\retour\services\Redirects->doRedirect('https://deathpe...', '/documents/ESPY...', Array)
#2 /home/forge/deathpenaltyinfo.org/vendor/nystudio107/craft-retour/src/Retour.php(514): nystudio107\retour\services\Redirects->handle404()
#3 [internal function]: nystudio107\retour\Retour->nystudio107\retour\{closure}(Object(craft\events\ExceptionEvent))
#4 /home/forge/deathpenaltyinfo.org/vendo
```

**Comment by khalwat** [comment]
@dlindberg fixed in https://github.com/nystudio107/craft-retour/commit/6ee10a6df8033d183a42dd07fea2087767fbd428

Can you set your semver to `dev-develop` and do a `composer update` to verify the fix works for you?

**Comment by khalwat** [comment]
Verified as fixed in 3.1.36
